packages/focusvisible/src/useFocusVisible.js

```diff
@@ -0,0 +1,255 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+/**
+ * Inspired by the [WICG/focus-visible](https://github.com/WICG/focus-visible)
+ * `:focus-visible` polyfill.
+ */
+
+import { useRef, useCallback, useEffect } from 'react';
+
+const INPUT_TYPES_WHITE_LIST = {
+  text: true,
+  search: true,
+  url: true,
+  tel: true,
+  email: true,
+  password: true,
+  number: true,
+  date: true,
+  month: true,
+  week: true,
+  time: true,
+  datetime: true,
+  'datetime-local': true
+};
+
+export function useFocusVisible({
+  scope,
+  relativeDocument = document,
+  className = 'garden-focus-visible',
+  dataAttribute = 'data-garden-focus-visible'
+} = {}) {
+  if (!scope) {
+    throw new Error('Error: the useFocusVisible() hook requires a "scope" property');
```

Comment:
I originally had this, but `ref` is a reserved keyword when used as a render-prop container.

This would make the API difficult to grok from the `FocusVisibleContainer` side.